### PR TITLE
feature(ui-ux): add nav menu

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- To add nav menu to desktop and mobile view
- To add temporary 3d model animation
- To add 3rd party library plugin to animate 3d model

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
- Currently using arbitrary model to replace jellyfish because we haven't catch one yet
- Component in NavMenu couldn't be "cleanly" refactored due to the arbitrary value for `content` property of `::before` and `::after` pseudo element cannot accept react variables. And the reason to use pseudo elements is because we have to create the transition effect of gradient text or gradient background

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*

### CODE REVIEW

File: src/components/home/Header.tsx [hunk 1]
```diff
@@ -1,35 +1,62 @@
...
-          <div className="relative w-[149px]">
-            <Image
-              data-testid="dmc_logo"
-              layout="fill"
-              src="/logo.svg"
-              alt="DeFi Meta Chain Logo"
-              objectFit="contain"
+    <section className="w-full h-screen relative bg-no-repeat bg-cover bg bg-[url('/portal-gradient-1.png')] px-4 pt-6 md:px-[40px] md:pt-6 lg:px[120px] lg:pt-[52px]">
+      <header>
+        <nav className="flex justify-between w-full items-center">
```
Comment (thedoublejay) [inline src/components/home/Header.tsx:14]: I think there's a position "top" for nav?

File: src/components/home/NavMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+  label,
+}: {
+  href: string | UrlObject;
+  imagePath: string;
+  altLabel?: string;
+  label?: React.ReactNode | string;
+}): JSX.Element {
+  return (
+    <Link href={href}>
+      <div className="py-3 px-5 flex items-center">
```
Comment (thedoublejay) [inline src/components/home/NavMenu.tsx:495]: Better to have "a" element here. Better for accessibility. Also, when you hover youy can see the links. You can also use `_blank` as default if it's not an in page navigation.

File: src/components/home/NavMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+          label={
+            <span
+              className={`${navChildItemStyle} before:content-['Releases'] after:content-['Releases']`}
+            >
+              Releases
+            </span>
+          }
+        />
+        <NavMenuChildItem
+          href=""
```
Comment (thedoublejay) [inline src/components/home/NavMenu.tsx:50]:
```suggestion
          href="https://github.com/DeFiCh/metachain"
```

File: src/components/home/NavMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+          label={
+            <span
+              className={`${navChildItemStyle} before:content-['Documentation'] after:content-['Documentation']`}
+            >
+              Documentation
+            </span>
+          }
+        />
+        <NavMenuChildItem
+          href=""
```
Comment (thedoublejay) [inline src/components/home/NavMenu.tsx:38]:
```suggestion
          href="https://github.com/DeFiCh/metachain/releases"
```

File: src/components/home/NavMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+          label={
+            <span
+              className={`${navChildItemStyle} before:content-['DMC_Explorer'] after:content-['DMC_Explorer']`}
+            >
+              DMC Explorer
+            </span>
+          }
+        />
+        <NavMenuChildItem
+          href=""
```
Comment (thedoublejay) [inline src/components/home/NavMenu.tsx:97]:
```suggestion
          href="https://defichain.com"
```

File: src/components/home/NavMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+          label={
+            <span
+              className={`${navChildItemStyle} before:content-['DeFiChain.com'] after:content-['DeFiChain.com']`}
+            >
+              DeFiChain.com
+            </span>
+          }
+        />
+        <NavMenuChildItem
+          href=""
```
Comment (thedoublejay) [inline src/components/home/NavMenu.tsx:109]:
```suggestion
          href="https://defichain.com/white-paper"
```

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: src/components/home/NavMenu.tsx (hunk 6 above)
Comment (kyleleow) [reply src/components/home/NavMenu.tsx:109]: i think we have a separate white paper for DMC

File: src/components/home/Header.tsx (hunk 1 above)
Comment (kyleleow) [reply src/components/home/Header.tsx:14]: do you mean the nav will be fixed on top upon scrolling down? Confirmed with designer last time that they won't be fixed

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**